src/DataStore.js

```diff
@@ -1302,6 +1303,15 @@ const props = {
               safeSetLink(record, inverseLocalField, this)
             } else {
               // Unset locals
+              if (current) {
+                console.log("SAFE SET FOREIGN KEY\n", current, foreignKey, "\n")
+                // Update (unset) inverse relation
```

Comment:
Formerly the above `// Update (unset) inverse relation` would only get called on update not on `clear/delete`. I added `safeSetLink(current, inverseLocalField, undefined)` which fixes the broken destroy test. 

On the other-hand attempting to add `safeSetProp(current, foreignKey, undefined)` (commented out below) which should make sense (Why would we keep foreign keys after clearing the relation?) causes 4 other specs to break in the collection_methods suite.